### PR TITLE
Enhanced Feature table to support 'always_enabled' value for state and auto-restart fields.

### PR DESCRIPTION
**Pull request by abdosi**
Why I did:
Fixes Issue#https://github.com/Azure/sonic-buildimage/issues/5999

How I did:

- Added new flag value 'always_enabled' for the state and auto-restart field of feature table

- init_cfg.json is updated to initialize state field of database/swss/syncd/teamd feature and auto-restart field of database feature 
  as always_enabled 

- Once the state/auto-restart value is initialized as "always_enabled" it is immutable and cannot be change via feature config commands. (config feature..) PR#https://github.com/Azure/sonic-utilities/pull/1271

- hostcfgd will not take any action if state field value is 'always_enabled'

- Since we have always_enabled field for auto-restart updated supervisor-proc-exit-listener
   not to have special check for database and always rely  on value from Feature table.

How I Verify:

- Cli command

```
admin@str-xxx-2:/var/log$ show feature status
Feature     State           AutoRestart
----------  --------------  --------------
acms        enabled         enabled
bgp         enabled         enabled
database    always_enabled  always_enabled
dhcp_relay  enabled         enabled
lldp        enabled         enabled
pmon        enabled         enabled
radv        enabled         enabled
snmp        enabled         enabled
swss        always_enabled  enabled
syncd       always_enabled  enabled
teamd       always_enabled  enabled
telemetry   enabled         enabled

```

- Hostcfgd skip the processing via syslog

`Nov 25 01:02:44.925773 str-xxx-2 INFO hostcfgd: Feature 'teamd' service is always enabled`

- Made sure auto-restart works fine for swss/syncd/teamd

### CODE REVIEW

**Comment by abdosi** [comment]
vsimage test "test_po_cleanup.py" failure is expected since we need to update the test case based on this change.
In the test case we are using config feature teamd disabled which will not work after this change.

**Review by rlhui — APPROVED** [review]

File: files/scripts/supervisor-proc-exit-listener [hunk 1]
```diff
@@ -77,27 +77,26 @@ def main(argv):
...
+            restart_feature = features_table[container_name].get('auto_restart')
+            if not restart_feature:
+                syslog.syslog(syslog.LOG_ERR, "Unable to determine auto-restart feature status for '{}'. Exiting...".format(container_name))
+                sys.exit(4)
+
+            # If auto-restart feature is not disabled and at the same time
             # a critical process exited unexpectedly, terminate supervisor
-            if ((container_name == 'database' or restart_feature == 'enabled') and expected == 0 and
-                    (processname in critical_process_list or groupname in critical_group_list)):
+            if ((restart_feature != 'disabled') and expected == 0 and
```
Comment (jleveque) [inline files/scripts/supervisor-proc-exit-listener:98]: Extra parentheses are no longer needed around `restart_feature != 'disabled'`

**Review by jleveque — CHANGES_REQUESTED** [review]
As comments.

Also please update PR title and description to align with new solution.

File: files/scripts/supervisor-proc-exit-listener (hunk 1 above)
Comment (abdosi) [reply files/scripts/supervisor-proc-exit-listener]: @jleveque Updated

**Review by jleveque — APPROVED** [review]

**Pull request merged**